image_extension.md

```diff
@@ -86,16 +86,51 @@ Correspondingly, each `/bin/generate` executable:
 - MAY read the Buildpack Plan.
 - MAY log output from the build process to `stdout`.
 - MAY emit error, warning, or debug messages to `stderr`.
-- MAY write a run.Dockerfile to the `<output>` directory. This file MUST adhere to the requirements listed below.
+- MAY write either or both of `build.Dockerfile` and `run.Dockerfile` to the `<output>` directory. This file MUST adhere to the requirements listed below.
+- MAY write key-value pairs to `<output>/extend-config.toml` that are provided as build args to build.Dockerfile when extending the build image.
 - MUST NOT write SBOM (Software-Bill-of-Materials) files as described in the [Software-Bill-of-Materials](#software-bill-of-materials) section.
 
 #### Dockerfile Requirements
 
-run.Dockerfiles:
+A `run.Dockerfile`
 
 - MAY contain a single `FROM` instruction
 - MUST NOT contain any other instructions
 
+A `build.Dockerfile`
+
+- MUST begin with:
+```bash
+ARG base_image
+FROM ${base_image}
+```
+- MUST NOT contain any other `FROM` instructions
+- MAY contain `ADD`, `ARG`, `COPY`, `ENV`, `LABEL`, `RUN`, `SHELL`, `USER`, and `WORKDIR` instructions
+- MUST NOT contain any other instructions
```

Comment:
I wonder if this should be `SHOULD NOT`, so that we could allow other commands, but not promise that they'll work. For example, someone might want to have an ENTRYPOINT so that they can run an image build from this Dockerfile, but they're fine with it getting ignored/overwritten during buildpack execution.